bork/log.py

```diff
@@ -0,0 +1,33 @@
+import inspect
+import logging
+from typing import Any, Callable, Optional, TypeVar
+
+
+def logger(context: Optional[inspect.FrameInfo] = None) -> logging.Logger:
+    """Provide a logger with a name appropriate for a given context.
+
+    The default context is the caller's.
+    """
+    if context is None:
+        context = inspect.getframeinfo(inspect.currentframe().f_back)
+
+    return logging.getLogger('bork.{}'.format(context.function))
+
+
+F = TypeVar('F', bound=Callable[..., Any])  # noqa
+
+
+def trace(func: F, level: int = logging.DEBUG) -> F:
+    """Decorator to log function entry and exit."""
+    log = logging.getLogger(__name__ + func.__name__)
+
+    def wrapper(*args, **kwargs):
+        arglist = args + ['{}={}'.format(k, v) for k, v in kwargs]
```

Comment:
Need to coerce everything to strings (i suggest `repr()`)

Reply:
Thanks :purple_heart: